components/script/dom/mouseevent.rs

```diff
@@ -192,6 +203,56 @@ impl MouseEventMethods for MouseEvent {
         }
     }
 
+    // https://drafts.csswg.org/cssom-view/#dom-mouseevent-x
+    fn X(&self) -> i32 {
+        self.client_x.get()
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-mouseevent-y
+    fn Y(&self) -> i32 {
+        self.client_y.get()
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-mouseevent-offsetx
+    fn OffsetX(&self) -> i32 {
+        let event = self.upcast::<Event>();
+        if event.dispatching() {
+            match event.GetTarget() {
+                Some(target) => {
+                    if let Some(element) = target.downcast::<Element>() {
```

Comment:
If we downcast to Node instead then we can call `Node::client_rect`. This means we should get expected behaviour for mouse events on non-Elements like text nodes, for example.